x/evm/keeper/integration_test.go

```diff
@@ -1,246 +1,252 @@
 )
 
-var _ = Describe("Feemarket", func() {
-	var privKey *ethsecp256k1.PrivKey
+type IntegrationTestSuite struct {
+	network     network.Network
+	factory     factory.TxFactory
+	grpcHandler grpc.Handler
+	keyring     testkeyring.Keyring
+}
 
-	Describe("Performing EVM transactions", func() {
-		type txParams struct {
-			gasLimit  uint64
-			gasPrice  *big.Int
-			gasFeeCap *big.Int
-			gasTipCap *big.Int
-			accesses  *ethtypes.AccessList
+var _ = Describe("Handling a MsgEthereumTx message", Label("EVM"), Ordered, func() {
+	var s *IntegrationTestSuite
+
+	BeforeAll(func() {
+		keyring := testkeyring.New(3)
+		integrationNetwork := network.New(
+			network.WithPreFundedAccounts(keyring.GetAllAccAddrs()...),
+		)
+		grpcHandler := grpc.NewIntegrationHandler(integrationNetwork)
+		txFactory := factory.New(integrationNetwork, grpcHandler)
+		s = &IntegrationTestSuite{
+			network:     integrationNetwork,
+			factory:     txFactory,
+			grpcHandler: grpcHandler,
+			keyring:     keyring,
 		}
-		type getprices func() txParams
+	})
+
+	AfterEach(func() {
+		// Start each test with a fresh block
+		err := s.network.NextBlock()
+		Expect(err).To(BeNil())
+	})
+
+	When("the params have default values", Ordered, func() {
+		BeforeAll(func() {
+			// Set params to default values
+			defaultParams := evmtypes.DefaultParams()
+			err := s.network.UpdateEvmParams(defaultParams)
+			Expect(err).To(BeNil())
+
+			err = s.network.NextBlock()
+			Expect(err).To(BeNil())
+		})
 
-		Context("with MinGasPrices (feemarket param) < BaseFee (feemarket)", func() {
-			var (
-				baseFee      int64
-				minGasPrices int64
+		It("performs a transfer transaction", func() {
+			senderPriv := s.keyring.GetPrivKey(0)
+			receiver := s.keyring.GetKey(1)
+			txArgs := evmtypes.EvmTxArgs{
+				To:     &receiver.Addr,
+				Amount: big.NewInt(1000),
+			}
+
+			res, err := s.factory.ExecuteEthTx(senderPriv, txArgs)
+			Expect(err).To(BeNil())
+			Expect(res.IsOK()).To(Equal(true), "transaction should have succeeded", res.GetLog())
+		})
+
+		It("performs a contract deployment and contract call", func() {
+			senderPriv := s.keyring.GetPrivKey(0)
+			constructorArgs := []interface{}{"coin", "token", uint8(18)}
+			compiledContract := contracts.ERC20MinterBurnerDecimalsContract
+			contractAddr, err := s.factory.DeployContract(
+				senderPriv,
+				compiledContract,
+				constructorArgs...,
 			)
+			Expect(err).To(BeNil())
+			Expect(contractAddr).ToNot(Equal(common.Address{}))
+
+			err = s.network.NextBlock()
+			Expect(err).To(BeNil())
+
+			txArgs := evmtypes.EvmTxArgs{
+				To: &contractAddr,
+			}
+			callArgs := factory.CallArgs{
+				ContractABI: compiledContract.ABI,
+				MethodName:  "mint",
+				Args:        []interface{}{s.keyring.GetAddr(1), big.NewInt(1e18)},
+			}
+			res, err := s.factory.ExecuteContractCall(senderPriv, txArgs, callArgs)
+			Expect(err).To(BeNil())
+			Expect(res.IsOK()).To(Equal(true), "transaction should have succeeded", res.GetLog())
```

Comment:
Add checks:
- we should check the logs for events
- we should check that the balance has been updated
- we should check that the supply has been updated